File: BinaryTree/csharp/TreeIncludes.cs
using System.Collections.Generic;

namespace BinaryTreeSolutions;

public static class TreeIncludes
{
    public static bool Solve(TreeNode? root, int target)
    {
        if (root is null)
        {
            return false;
        }

        var queue = new Queue<TreeNode>();
        queue.Enqueue(root);
        while (queue.Count > 0)
        {
            var current = queue.Dequeue();
            if (current.Val == target)
            {
                return true;
            }

            if (current.Left is not null) queue.Enqueue(current.Left);
            if (current.Right is not null) queue.Enqueue(current.Right);
        }

        return false;
    }
}

File: BinaryTree/ts/treeIncludes.ts
import { TreeNode } from "./TreeNode";

export function treeIncludes(root: TreeNode<number> | null, target: number): boolean {
  if (root === null) {
    return false;
  }

  const queue: TreeNode<number>[] = [root];
  while (queue.length > 0) {
    const current = queue.shift()!;
    if (current.val === target) {
      return true;
    }
    if (current.left) queue.push(current.left);
    if (current.right) queue.push(current.right);
  }

  return false;
}
